src/system-update-generator/system-update-generator.c

```diff
@@ -18,21 +19,22 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
+        FOREACH_STRING(p, "/system-update", "/etc/system-update") {
+                if (laccess(p, F_OK) < 0) {
+                        if (errno == ENOENT)
+                               continue;
 
-        if (laccess("/system-update", F_OK) < 0) {
-                if (errno == ENOENT)
-                        return 0;
+                        return log_error_errno(errno, "Failed to check for system update: %m");
```

Comment:
if you try multiple files anyway i'd probably just warn about unexpected errors, and continue the loop.

i.e.

```c
_cleanup_free_ char *j = NULL;

FOREACH_STRING(p, "/system-update", "/etc/system-update") {
        if (laccess(p, F_OK) >= 0) 
                break;
        if (errno != ENOENT)
                log_warning_errno(errno, "Failed to check for system-update flag file, ignoring: %m");
}
if (!p)
        return 0;

j = path_join(arg_dest, SPECIAL_DEFAULT_TARGET);
…
```

Reply:
Just re-pushed, I basically used this suggested code with a `goto link_found` instead of the `if (!p)` condition, because p is out of scope at this point.